src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,103 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+const newMoviesDefaultFields = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [newMovies, setNewMovies] = useState(newMoviesDefaultFields);
+
+  const handleChange = (name: string, value: string) => {
+    setNewMovies(prevstate => ({
+      ...prevstate,
+      [name]: value,
+    }));
+  };
+
+  const resetMoviesForm = () => {
+    setCount(prevcaunt => prevcaunt + 1);
+    setNewMovies(newMoviesDefaultFields);
+  };
+
+  const isNotValid =
+    newMovies.title.trim() === '' ||
```

Comment:
```suggestion
    !newMovies.title.trim() ||
```
it will look more laconic